applications/CoSimulationApplication/python_scripts/data_transfer_operators/kratos_mapping_with_debug.py

```diff
@@ -0,0 +1,379 @@
+        self.number_of_mappers = 0
+
+    def _ExecuteTransferData(self, from_solver_data, to_solver_data, transfer_options):
+        """
+        Executes the transfer of data between solvers.
+
+        Args:
+            from_solver_data (CoSimulationData): The data from the source solver to transfer.
+            to_solver_data (CoSimulationData): The data for the target solver to receive the transfer.
+            transfer_options (KM.Flags): Options that specify how the transfer should be executed.
+
+        Returns:
+            None
+
+        This method prepares the solver data for transfer, checks the available mappers, and executes 
+        the transfer process. If the corresponding mappers for the given identifiers are not found, 
+        it creates a new mapper based on the distribution of the model parts and then executes the mapping. 
+        It also generates VTK output for debugging purposes based on the auxiliary debug counter setting.
+        """
+        # Prepare the solver data
+        solver_data = self._PrepareSolverData(from_solver_data, to_solver_data, transfer_options)
+
+        if solver_data.identifier_tuple in self._mappers:
+            # Generate VTK output for debugging
+            if self.auxiliary_debug_counter:
+                self.__GenerateProcessVTK(from_solver_data, to_solver_data, transfer_options)
+            self.__PostProcessVTKPre(solver_data.identifier_tuple, from_solver_data, to_solver_data, transfer_options)
+            self._mappers[solver_data.identifier_tuple].Map(solver_data.variable_origin, solver_data.variable_destination, solver_data.mapper_flags)
+            self.__PostProcessVTKPost(solver_data.identifier_tuple, from_solver_data, to_solver_data, transfer_options)
+        elif solver_data.inverse_identifier_tuple in self._mappers:
+            # Generate VTK output for debugging
+            if self.auxiliary_debug_counter:
+                self.__GenerateProcessVTK(from_solver_data, to_solver_data, transfer_options)
+            self.__PostProcessVTKPre(solver_data.inverse_identifier_tuple, from_solver_data, to_solver_data, transfer_options)
+            self._mappers[solver_data.inverse_identifier_tuple].InverseMap(solver_data.variable_destination, solver_data.variable_origin, solver_data.mapper_flags)
+            self.__PostProcessVTKPost(solver_data.inverse_identifier_tuple, from_solver_data, to_solver_data, transfer_options)
+        else:
+            # Generate VTK output for debugging
+            self.__GenerateProcessVTK(from_solver_data, to_solver_data, transfer_options)
+
+            # Get the model parts
+            model_part_origin      = self._GetModelPartFromInterfaceData(from_solver_data)
+            model_part_destination = self._GetModelPartFromInterfaceData(to_solver_data)
+
+            if model_part_origin.IsDistributed() or model_part_destination.IsDistributed():
+                mapper_create_fct = python_mapper_factory.CreateMPIMapper
```

Comment:
why do you override the creation of mappers? 
How about calling the baseclass for this instead?